contracts/pallet-ibc/src/ics20/mod.rs

```diff
@@ -561,11 +563,370 @@ pub fn full_ibc_denom(packet: &Packet, mut token: PrefixedCoin) -> String {
 use ibc::applications::transfer::error::Error as Ics20Error;
 
 pub trait HandleMemo<T: Config> {
-	fn execute_memo(packet_data: &PacketData, receiver: T::AccountId) -> Result<(), Ics20Error>;
+	fn execute_memo(packet: &Packet) -> Result<(), Ics20Error>;
```

Comment:
why is the receiver no longer important?

Reply:
You asked [here](https://github.com/ComposableFi/centauri/pull/359#discussion_r1245079395) to merge everything together and pass in execute_memo trait function.
So execute memo receives packet and then check memo and take receiver account to process memo.